### PR TITLE
#4820 Faces 3.0 Does Not Support JavaEE NameSpace in Flow.xml Files

### PR DESCRIPTION
**Pull request by hussainnm**
Introduce a constructor to use given namespace instead of default

### CODE REVIEW

**Review by manorrock — COMMENTED** [review]
Can you add some tests?

**Comment by hussainnm** [comment]
Added Tests

**Review by manorrock — COMMENTED** [review]
Can you also create a PR against the master branch? Please refer to the issue in its comments.

File: impl/src/main/java/com/sun/faces/config/processor/FacesConfigNamespaceContext.java [hunk 1]
```diff
@@ -23,9 +23,19 @@
 
 class FacesConfigNamespaceContext implements NamespaceContext {
 
+    String defaultNamespace = null;
```
Comment (pzygielo) [inline impl/src/main/java/com/sun/faces/config/processor/FacesConfigNamespaceContext.java:26]: Could it be private?

**Comment by arjantijms** [comment]
> Can you also create a PR against the master branch? Please refer to the issue in its comments.

We can additionally cherry pick, it's no super big deal, though PR may be more convenient ;)

File: impl/src/main/java/com/sun/faces/config/processor/FacesConfigNamespaceContext.java [hunk 2]
```diff
@@ -23,9 +23,19 @@
...
+
+    public FacesConfigNamespaceContext() {
```
Comment (pzygielo) [inline impl/src/main/java/com/sun/faces/config/processor/FacesConfigNamespaceContext.java:28]: I think extending accessibility of that constructor (current (existing) default one isn't public) is not necessary. Given type is package-accessible anyway.

**Review by manorrock — APPROVED** [review]

**Pull request merged**